Initiate merged report and write intro

allyhawkins opened the pull request:
Closes #624 

This PR takes the previous integration report and renames it to `merge-report.Rmd`. I also updated the parameters to be consistent with our use of `merge_group` and `merge_sce`. 

In the actual report itself, I removed any integration specific steps or text. I added a nice little yellow warning box at the beginning. Then I included some introductory text. I also have it printing out a table temporarily, but that will get removed as part of #625. 

- What do we think of the text in the warning box? Do we want to say anything else in the box?
- Does the intro text cover what we would expect to be there based on our discussion and the description in the issue. 

Here's a screenshot of what it looks like right now: 
<img width="792" alt="Screenshot 2023-12-19 at 4 06 02 PM" src="https://github.com/AlexsLemonade/scpca-nf/assets/54039191/48565fb6-3336-444e-bc63-b5e3ca681275">

sjspielman commented on templates/merge-report.Rmd:
```diff
@@ -0,0 +1,82 @@
+  batch_column: "library_id"
+
+title: "`r glue::glue('Merge report for {params$merge_group}')`"
+author: "CCDL"
```
To match the other reports:

```suggestion
author: "Childhood Cancer Data Lab"
```

sjspielman commented on templates/merge-report.Rmd:
```diff
@@ -0,0 +1,82 @@
+  merged_sce: NULL
+  batch_column: "library_id"
+
+title: "`r glue::glue('Merge report for {params$merge_group}')`"
```
```suggestion
title: "`r glue::glue('ScPCA Merged report for {params$merged_group}')`"
```

sjspielman commented on templates/merge-report.Rmd:
````diff
@@ -0,0 +1,82 @@
+    code_download: true
+---
+
+```{r, echo=FALSE, results='asis'}
````
Let's define this chunk as unevaluated with a label, and then call it with `eval=TRUE` in the 2 spots where we'll need it.
Reference: https://bookdown.org/yihui/rmarkdown-cookbook/reuse-chunks.html#same-label

sjspielman commented on templates/merge-report.Rmd:
```diff
@@ -0,0 +1,82 @@
+glue::glue("
+    <div class=\"alert alert-warning\">
+
+    Caution: The libraries included in the merged object are NOT integrated or batch-corrected.
```
More yelling (I think `<b>` should work for bold, but please check!)

```suggestion
    <b>CAUTION: The libraries included in the merged object are NOT integrated or batch-corrected.</b>
```

sjspielman commented on templates/merge-report.Rmd:
```diff
@@ -0,0 +1,82 @@
+This report includes a brief summary of all libraries included in the merged object for `r params$merge_group`.
+This merged object is intended to allow for downstream analyses across multiple samples.
+However, the merged object has NOT been integrated or batch-corrected. 
+To conduct any downstream analyses that require batch correction (e.g., UMAP or clustering), batch correction must be calculated separately. 
```
```suggestion
To conduct any downstream analyses that require batch correction (e.g., UMAP or clustering), batch correction must be performed separately. 
```

sjspielman commented on templates/merge-report.Rmd:
```diff
@@ -0,0 +1,82 @@
+
+
+This report includes a brief summary of all libraries included in the merged object for `r params$merge_group`.
+This merged object is intended to allow for downstream analyses across multiple samples.
```
```suggestion
This merged object is intended to facilitate for downstream analyses across multiple samples.
```

sjspielman commented on templates/merge-report.Rmd:
````diff
@@ -0,0 +1,82 @@
+To conduct any downstream analyses that require batch correction (e.g., UMAP or clustering), batch correction must be calculated separately. 
+
+
+```{r setup, message = FALSE, echo = FALSE}
+# knitr options
+knitr::opts_chunk$set(
+  echo = FALSE
+)
+
+library(SingleCellExperiment)
+library(ggplot2)
+
+# Set default ggplot theme
+theme_set(
+  theme_bw() +
+    theme(plot.margin = margin(rep(20, 4)))
+)
+```
+
+```{r}
+# check that input files are provided
+if (is.null(params$merged_sce)) {
+  stop("Must provide a merged SCE object.")
+}
+
+# define batch column
+batch_column <- params$batch_column
+```
````
Can we move these lines up to be right after the YAML? We should define the `setup` chunk first, and we'll fail a smidge faster.

sjspielman commented on merge.nf:
```diff
@@ -159,8 +158,8 @@ workflow {
-    // TODO: generate merge report
-    //merge_report(merge_sce.out, file(merge_template))
+    // generate merge report
+    merge_report(merge_sce.out, file(merge_template))
```
We're not really merging reports here, so...
```suggestion
    generate_merge_report(merge_sce.out, file(merge_template))
```

sjspielman replied:
And also the definition on L60

sjspielman commented on templates/merge-report.Rmd:
```diff
@@ -0,0 +1,82 @@
+
+This report includes a brief summary of all libraries included in the merged object for `r params$merge_group`.
+This merged object is intended to allow for downstream analyses across multiple samples.
+However, the merged object has NOT been integrated or batch-corrected. 
```
Let's add a new line for emphasis so this text doesn't get too buried, with some italics for more yelling? Is this too much yelling..? Is there such a thing as too much yelling here?

FYI I'm also removing `However` because this isn't really a contradiction of the previous statement, just another statement!

```suggestion

_The libraries in this merged object have not been integrated or batch-corrected._
```

sjspielman replied:
Should we actually make this directly match the yellow box with `"included in"` instead of just `"in"`? I'm not sure it matters much. But, I do think we should refer to the libraries and not the object itself.

allyhawkins commented:
- I updated some of the language per your suggestions. 
- I moved the set up chunks to be above the warning. 
- The bold in the warning works, so I kept that in. 
- For the warning chunk, I added a label and then reference that label to evaluate it. 

This should be ready for another review.

sjspielman reviewed (approved):
🚀
